tests/instruments/test_all_instruments.py

```diff
@@ -78,6 +78,8 @@ def test_adapter_arg(cls):
 # Instruments which do not yet accept "name" argument
 nameless_instruments = [
     "AdvantestR3767CG",
+    "AdvantestR6245",
+    "AdvantestR6246",
```

Comment:
When running pytest, the name variable is passed through kwargs (see error below). Thus the solution you presented above doesn't work. Any suggestions?

`TypeError: pymeasure.instruments.advantest.advantestR624X.AdvantestR624X.__init__() got multiple values for keyword argument 'name'_`

Reply:
You have to add a name parameter to the main class as well, similar to the sub classes.

Reply:
This has no effect as the error is raised in the parent constructor (AdvantestR6245).

The error is raised because the constructor is called like this:
`AdvantestR6246(name="Advantest R6246 SourceMeter", kwargs = {'name': 'Name_Test'})`

Reply:
You have to make it an explicit keyword argument: `name="R6X Source meter Base Class" `. In this case, the kwargs replace the keyword name.

For that reason, it works with the 45 and 46 instruments.

I'm sorry, that I was not clear enough.

Reply:
Thank you for your fast replies! The solutions a suggested is not working for me. I've done an intermediate commit to show you that I did as you instructed.

```
FAILED tests/instruments/test_all_instruments.py::test_name_argument[AdvantestR6245] - TypeError: pymeasure.instruments.advantest.advantestR624X.AdvantestR624X.__init__() got multiple values for keyword argument 'name'
FAILED tests/instruments/test_all_instruments.py::test_name_argument[AdvantestR6246] - TypeError: pymeasure.instruments.advantest.advantestR624X.AdvantestR624X.__init__() got multiple values for keyword argument 'name'
=
```

Reply:
At least the CI tests pass now.

I have to take look tomorrow.

Reply:
If all `Instrument` based classes have `def __init__(self, adapter, name="abc", **kwargs):`, the test should pass (other instruments do pass as well). Right now, the tests succeed. Could you check, whether the tests still fail for you?

Thanks!

Reply:
Ouch, pytest was using an old instance of my instrument class, and thus failing. This issue is now fixed. Sorry for the extra time you spend on this.